tasks_week3_Shai_Axelrod/Zoo/include/Animal.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include "Location.h"
+#include "Direction.h"
+/**
+ * @class Animal
+ * @brief Represents an animal in the simulation.
+ */
+class Animal {
+protected:
+    int steps; /**< The number of steps the takes in each move. */
+    std::string name; /**< The name of the animal. */
+    Location location; /**< The current location of the animal. */
+    bool inMoveMode; /**< Whether the animal is currently in move mode. */
+    Direction direction; /**< The direction the animal is facing. */
+
+    /**
+     * @brief Returns the type of the animal.
+     * @return The type of the animal as a string.
+     */
+    virtual std::string getType() const = 0;
+
+    /**
+     * @brief Randomly selects a new direction for the animal to move.
+     */
+    virtual void ballotNewDir() = 0;
+
+public:
+    /**
+     * @brief Constructor for the Animal class.
+     * @param name The name of the animal.
+     * @param location The initial location of the animal.
+     */
+    Animal(const std::string& name, const Location& location);
+
+    /**
+     * @brief Prints the details of the animal.
+     */
+    void printDetails() const;
+
+    /**
+     * @brief Returns the initial of the animal's name.
+     * @return The initial of the animal's name as a character.
+     */
+    char getInitial() const;
+
+    /**
+     * @brief Returns the current location of the animal.
+     * @return The current location of the animal.
+     */
+    Location getLocation() const;
+
+    /**
+     * @brief Moves the animal one step in its current direction.
+     */
+    void step();
+
```

Comment:
no use of virtual 
read this for better understanding
https://stackoverflow.com/questions/2391679/why-do-we-need-virtual-functions-in-c

Reply:
the "step" operation - and as a result the step function implementation is common to all animals so no need to use virtual here.
(However, the step operation indeed be virtually at the end of the day - because the step is affected by the direction, which is determined virtually. for more details please see the step implementation and ballotNewDir virtual function)